shell/browser/api/electron_api_web_contents.cc

```diff
@@ -2007,18 +2023,64 @@ void WebContents::Stop() {
   web_contents()->Stop();
 }
 
+bool WebContents::CanGoBack() const {
+  return web_contents()->GetController().CanGoBack();
+}
+
 void WebContents::GoBack() {
+  if (!CanGoBack())
+    return;
   web_contents()->GetController().GoBack();
```

Comment:
nit: 
```suggestion
  if (CanGoBack())
    web_contents()->GetController().GoBack();
```

reads a bit easier. same for the other methods here.